recurly/__init__.py

```diff
@@ -1061,8 +1080,11 @@ def filter_currency(resources):
         except AttributeError:
             pass
 
+        return self.__request_invoice(url, self)
+
+    def __request_invoice(self, url, body=None):
```

Comment:
Doesn't this break the other calls? This maybe should be `body=self` or you should change the other calls to include a body that need it.

Reply:
It doesn't break the other calls, afaik. `invoice()`, `preview()`, `authorize()`, and `pending()` need `body` to equal `self` as you've suggested, which we accomplish by passing in `self` as the third argument  on line 1083. 

`capture()` and `cancel()`, however, require that `body=None`, so we accomplish that by setting the default value for `body` to `None` in `__request_invoice()`. `capture()` and `cancel()` then call `__request_invoice()` directly, not `__invoice()`.

I checked the to see if `preview()`, `authorize()` and `pending()` still sent `purchase` as a body, and they do.

Reply:
Joanna and I talked about the body issues earlier today and I suggested that she break the `__invoice` method up and extract the part that sends and parses the response into it's own method.

The idea being that the `__invoice` method just pass `self` for the body param and the `capture` and `cancel` can call `__request_invoice` directly without the body

Reply:
> invoice(), preview(), authorize(), and pending() need body to equal self as you've suggested, which we accomplish by passing in self as the third argument on line 1083.

Ah okay, I missed that change. It makes more sense now that i've expanded the diff.